src/pages/jam/Jam.svelte

```diff
@@ -197,67 +186,62 @@
         }
     }
 
-    function connectWS() {
-        if (jam) {
-            JamStore.set({
-                ...jam,
-                ws: new WebSocket(`${WS_BASE_URL}/jam/${jam.id}`),
+    function initJam() {
+        return api
+            .get<GetJamData>(`/jam/${jamID}`)
+            .then(({ data }) => {
+                JamStore.set({
+                    ...data,
+                    players: [],
+                    ws: new WebSocket(`${WS_BASE_URL}/jam/${jamID}`),
+                });
+                Success('Jam data loaded');
+            })
+            .catch((err: AxiosError) => {
+                Failure(err.message);
+                navigate('/', { replace: true });
             });
-            return;
-        }
-
-        Failure('Jam not found');
-        if (jam && jam.ws) {
-            jam.ws.close();
-        }
-
-        navigate('/', { replace: true });
     }
 
     function sendWSMsg(msg: WSMsg<any>) {
-        jam.ws.send(JSON.stringify(msg));
+        $JamStore.ws.send(JSON.stringify(msg));
     }
 
     function handleWSMsg(msg: WSMsg<MIDIMsg | string>) {
         switch (msg.type) {
             case WSMsgTyp.TEXT:
-                JamTextStore.set([...textMsgs, msg.payload as string]);
+                JamTextStore.update((items) => [
+                    ...items,
+                    msg.payload as string,
+                ]);
                 break;
             case WSMsgTyp.MIDI:
                 midiDiv.scrollTop = midiDiv.scrollHeight;
-                JamMIDIStore.set([...midiMsgs, msg.payload as MIDIMsg]);
+                JamMIDIStore.update((items) => [
```

Comment:
I don't think we should store the MIDI messages. I think this will eventually slow down the app when there are many incoming messages from all the players. We should just immediately attempt to audibly play the message back, and drop it from memory